core/trino-main/src/main/java/io/trino/metadata/Metadata.java

```diff
@@ -114,6 +114,16 @@
 
     Optional<Object> getInfo(Session session, TableHandle handle);
 
+    /**
+     * Return table schema definition for the specified table handle.
+     * Table schema definition is a set of information
+     * required by semantic analyzer to analyze the query.
+     * This is method should be preferred over {@link #getTableMetadata(Session, TableHandle)} as it is cheaper.
```

Comment:
Instead of (or additionally to) having this sentence here, add a plain `@see` pointing to `getTableMetadata`
In `getTableMetadata` add `@see` poitining here and recommending use of this method, as it returns less information